charts/linkerd2/templates/heartbeat.yaml

```diff
@@ -39,9 +39,13 @@ spec:
             imagePullPolicy: {{.Values.global.imagePullPolicy}}
             args:
             - "heartbeat"
-            - "-prometheus-url=http://linkerd-prometheus.{{.Values.global.namespace}}.svc.{{.Values.global.clusterDomain}}:9090"
             - "-controller-namespace={{.Values.global.namespace}}"
             - "-log-level={{.Values.controllerLogLevel}}"
+            {{- if .Values.global.prometheusUrl }}
+            - "-prometheus-url={{.Values.global.prometheusUrl}}"
+            {{- else if .Values.prometheus.enabled }}
```

Comment:
Seems you'll have to leave this at just `else` for now for the integration tests to pass